RecoEgamma/EgammaTools/interface/AnyMVAEstimatorRun2Base.h

```diff
@@ -1,20 +1,26 @@
 #ifndef RecoEgamma_EgammaTools_AnyMVAEstimatorRun2Base_H
 #define RecoEgamma_EgammaTools_AnyMVAEstimatorRun2Base_H
 
-#include "FWCore/ParameterSet/interface/ParameterSet.h"
+// Note on Python/FWLite support: please forward declare as much as possible in
+// this header, because there are usecases of generating the dictionaries for
+// this class on the fly (see notes in ElectronMVAEstimatorRun2.h for more
+// details).
 
-#include "FWCore/Framework/interface/Event.h"
-#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include <string>
+#include <vector>
 
-#include "DataFormats/Candidate/interface/Candidate.h"
+namespace edm {
+  class ParameterSet;
+}  // namespace edm
```

Comment:
Replacing this with
```suggestion
#include "FWCore/Framework/interface/FrameworkfwdMostUsed.h"
```
would be slightly better.

Reply:
Cool, I didn't know about that!

Reply:
I guess it's a natural question now why not to use [CandidateFwd.h](https://github.com/cms-sw/cmssw/blob/master/DataFormats/Candidate/interface/CandidateFwd.h) for `reco::Candidate` then, but CandidateFwd.h includes way too much itself, to that's not an option.

Reply:
This comment was not implemented, but it can be deferred to a later clean-up PR

Reply:
Oh, I thought I addressed it but I think that change got lost in interactive rebasing I guess. I'll make that change another time then!